File: libnetwork/src/main/java/com/gystry/libnetwork/ApiService.java
package com.gystry.libnetwork;

import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;
import java.util.concurrent.TimeUnit;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSession;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * @author gystry
 * 创建日期：2020/7/3 11
 * 邮箱：gystry@163.com
 * 描述：
 */
public class ApiService {
    public static final OkHttpClient okHttpClient;
    public static String sBaseUrl;
    public static Convert sConvert;

    static {
        //日志拦截器
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        //创建okhttpClient实例
        okHttpClient = new OkHttpClient.Builder()
                .connectTimeout(5, TimeUnit.SECONDS)
                .writeTimeout(5, TimeUnit.SECONDS)
                .readTimeout(5, TimeUnit.SECONDS)
                .addNetworkInterceptor(httpLoggingInterceptor)
                .build();

        //配置网络信任所有的证书
        final TrustManager[] trustManagers = {new X509TrustManager() {
            @Override
            public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {

            }

            @Override
            public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {

            }

            @Override
            public X509Certificate[] getAcceptedIssuers() {
                return new X509Certificate[0];
            }
        }
        };
        final SSLContext ssl;
        try {
            ssl = SSLContext.getInstance("SSL");
            ssl.init(null,trustManagers,new SecureRandom());
            HttpsURLConnection.setDefaultSSLSocketFactory(ssl.getSocketFactory());
            HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
                @Override
                public boolean verify(String hostname, SSLSession session) {
                    return true;
                }
            });
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        } catch (KeyManagementException e) {
            e.printStackTrace();
        }

    }

    public static void init(String baseUrl,Convert convert){
        sBaseUrl = baseUrl;
        if (convert ==null) {
            convert =new JsonConvert();
        }
        sConvert = convert;
    }

    public static  <T> GetRequest<T> get(String url){
        return new GetRequest<>(sBaseUrl+url);
    }

    public static  <T> PostRequest<T> post(String url) {
        return new PostRequest<>(sBaseUrl + url);
    }
}

File: libnetworkkt/src/main/java/com/gystry/libnetworkkt/ApiService.kt
package com.gystry.libnetworkkt

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import java.security.SecureRandom
import java.security.cert.X509Certificate
import java.util.concurrent.TimeUnit
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

/**
 * @author gystry
 * 创建日期：2021/3/18 15
 * 邮箱：gystry@163.com
 * 描述：
 */
object ApiService {
    public val okHttpClient: OkHttpClient by lazy {
        OkHttpClient.Builder()
                .connectTimeout(5, TimeUnit.SECONDS)
                .writeTimeout(5, TimeUnit.SECONDS)
                .readTimeout(5, TimeUnit.SECONDS)
                .addNetworkInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
                .build()
    }
    public var sBaseUrl: String? = null
    public lateinit var sConvert: Convert<*>

    init {
        //配置网络信任所有的证书
        val trustManagers = arrayOf<TrustManager>(object : X509TrustManager {
            override fun checkClientTrusted(p0: Array<out X509Certificate>?, p1: String?) {

            }

            override fun checkServerTrusted(p0: Array<out X509Certificate>?, p1: String?) {

            }

            override fun getAcceptedIssuers(): Array<X509Certificate?> {
                return arrayOfNulls<X509Certificate>(0)
            }

        })

        var ssl: SSLContext? = null
        try {
            ssl = SSLContext.getInstance("SSL")
            ssl.init(null, trustManagers, SecureRandom())
            HttpsURLConnection.setDefaultSSLSocketFactory(ssl.socketFactory)
            HttpsURLConnection.setDefaultHostnameVerifier { _, _ -> true }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun init(baseUrl: String, convert: Convert<*>?) {
        var convertS = convert
        sBaseUrl = baseUrl
        if (convertS == null) {
            convertS = JsonConvert<Any>()
        }
        sConvert = convertS
    }


    //运算符重载
    operator fun <T> get(url: String): GetRequest<T> {
        return GetRequest(sBaseUrl + url)
    }

    fun <T> post(url:String):PostRequest<T>{
        return PostRequest(sBaseUrl+url)
    }

}

